module/x/gravity/keeper/pool.go

```diff
@@ -115,6 +130,11 @@ func (k Keeper) RemoveFromOutgoingPoolAndRefund(ctx sdk.Context, txId uint64, se
 	if err != nil {
 		return sdkerrors.Wrapf(types.ErrInvalid, "txId %d not in unbatched index! Must be in a batch!", txId)
 	}
+	// Make sure the tx was removed
```

Comment:
This check may be completely unnecessary, but it should be O(1) and only happens on unbatched transaction cancellation